src/dbnode/encoding/types.go

```diff
@@ -139,6 +142,8 @@ type Iterator interface {
 
 	// Close closes the iterator and if pooled will return to the pool.
 	Close()
+
+	SetSchema(descr namespace.SchemaDescr)
```

Comment:
Here too

Reply:
Ordering nit: I like to order the functions in interfaces according to lifecycle/order of use of at all possible. E.g start with Open, end with Close, etc. mind reordering took